src/main/java/org/jenkinsci/plugins/runselector/RunSelector.java

```diff
@@ -47,7 +47,7 @@
      * @throws IOException if an error occurs while performing the operation.
      * @throws InterruptedException if any thread interrupts the current thread.
      */
-    public Run<?, ?> pickBuildToCopyFrom(@Nonnull Job<?,?> job, @Nonnull final RunSelectorPickContext context)
+    public Run<?, ?> select(@Nonnull Job<?,?> job, @Nonnull final RunSelectorContext context)
```

Comment:
Good practice. Add annotations like CheckForNull in cases, where null results are the part of the common execution flow